clippy_lints/src/assign_ops.rs

```diff
@@ -135,6 +135,19 @@ impl LateLintPass for AssignOps {
                                         } else {
                                             return; // useless if the trait doesn't exist
                                         };
+                                        // check that we are not inside an `impl AssignOp` of this exact operation
+                                        let parent_fn = cx.tcx.map.get_parent(e.id);
+                                        let parent_impl = cx.tcx.map.get_parent(parent_fn);
+                                        if parent_impl.as_u32() != 0 {
```

Comment:
uuuhh... once we get `panic!` inside `const fn`, we can properly implement `NonZero::new` and then we can go mad on the microoptimizations (most relevant here: `Option<NonZero<NodeId>>` being the same size as `NodeId`).